src/server.js

```diff
@@ -64,39 +69,68 @@ const main = async () => {
     return new Response(200)
   }))
 
-  const eventSchema = Joi.string().required()
+  const eventSchema = Joi.string().pattern(/^\$/, { invert: true }).required()
   const propertiesSchema = Joi.object({
-    userId: Joi.string().required(),
-    appId: Joi.string().required(),
-    appPath: Joi.string().required(),
-    timestamp: Joi.date().timestamp().required()
-  }).required().unknown(true)
+    token: Joi.any().forbidden(),
+    'distinct_id': Joi.any().forbidden(),
+    time: Joi.any().forbidden(),
+    ip: Joi.any().forbidden(),
+    name: Joi.any().forbidden(),
+    appId: Joi.string().required()
+  }).pattern(Joi.string().pattern(/^(\$|mp_)/, { invert: true }), Joi.any().required()).required()
 
   /**
-   * @api {post} /event Log a user event
+   * @api {post} /api/event Log a user event
    * @apiDescription Records the event to a log and forwards it to mixpanel
    * @apiName event
    * @apiVersion 1.0.0
    * @apiGroup Events
    * @apiParam {String} event Name of the event
    * @apiParam {Object} properties Properties associated with this event. The below fields are required. Additional application defined fields can also be used
-   * @apiParam {String} properties.userId The anonymized ID of the user
    * @apiParam {String} properties.appId The application
-   * @apiParam {String} properties.appPath The navigational path in the application the user is on, with identifying parameters removed
-   * @apiParam {Date} properties.timestamp Timestamp of the event
    * @apiSuccess (Success 200) -
    */
   app.post('/api/event', promiseHandler(withAuth(async req => {
-    const { event, properties } = req.body
-
     validateInput(req.body, Joi.object({ event: eventSchema, properties: propertiesSchema }))
-
-    log(req.body)
-    token && sendToMixpanel(token, event, properties)
-
+    const data = _.update('properties', properties => ({
+      ...properties,
+      token,
+      'distinct_id': `google:${req.user.userSubjectId}`,
+      ip: req.ip
```

Comment:
Will this send the IP address of the user to mixpanel? If so we will need to remove this as it is considered identifiable information

Reply:
Ah, ok. I'll take that out.